### PR TITLE
stop using `Pkg.Types.Project`

### PR DESCRIPTION
**Pull request by KristofferC**
This allows registering weak dependencies to work on earlier Julia versions than 1.9. Unfortunately, there wasn't really any way to make this non-breaking, since the public API asks for a `Pkg.Types.Project` (and that doesn't even have the information to get the project file).

### CODE REVIEW

**Comment by KristofferC** [comment]
@GunnarFarneback, thoughts on this?

File: test/regedit.jl [hunk 1]
```diff
@@ -632,21 +623,21 @@ end
 # package. The registration is pushed to a new branch via a file URL.
 @testset "register" begin
     mktempdir(@__DIR__) do temp_dir
-        registry1_path = joinpath(temp_dir, "Registry1")
+        registry1_path = joinpath(@__DIR__, temp_dir, "Registry1")
```
Comment (GunnarFarneback) [inline test/regedit.jl:626]: Why this change? `mktempdir` returns an absolute path, doesn't it?

File: src/register.jl [hunk 2]
```diff
@@ -35,7 +35,7 @@ for the project file in that tree and a hash string for the tree.
 #
 #         # parse the project file
 #         @debug("parse the project file")
-#         project = Pkg.Types.read_project(project_file)
+#         project = Pkg.Types.project_file
```
Comment (GunnarFarneback) [inline src/register.jl:38]: What is `Pkg.Types.project_file`? Seems better to just remove this outcommented function.

File: src/register.jl [hunk 3]
```diff
@@ -732,29 +721,17 @@ function register(
     return set_metadata!(regbr, status)
 end
 
+# TODO: Move to Registrator.jl?
```
Comment (GunnarFarneback) [inline src/register.jl:724]: :+1:

File: src/register.jl [hunk 4]
```diff
@@ -614,7 +599,7 @@ function check_and_update_registry_files(pkg, package_repo, tree_hash,
     # update package data: compat file
     @debug("check compat section")
     regpaths = [registry_path; registry_deps_paths]
```
Comment (GunnarFarneback) [inline src/register.jl:601]: Let's get rid of this line too.

**Review by GunnarFarneback — APPROVED** [review]

**Comment by KristofferC** [comment]
Updated based on review.
